modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/SortAggregateIntegrationTest.java

```diff
@@ -114,21 +127,80 @@ public void mapReduceAggregate() {
         });
     }
 
+    /** */
+    @Test
+    public void correctCollationsOnMapReduceSortAgg() throws InterruptedException {
+        fillCacheTbl1(grid(0).cache("TEST"), ROWS);
+
+        QueryEngine engine = Commons.lookupComponent(grid(0).context(), QueryEngine.class);
+
+        List<FieldsQueryCursor<List<?>>> cursors = engine.query(
+            null,
+            "PUBLIC",
+            "SELECT /*+ DISABLE_RULE('HashAggregateConverterRule' ,'HashSingleAggregateConverterRule', " +
```

Comment:
Too complicated query. To reproduce the problem it's enough just two columns in the table (PK, COL0) and descending index by COL0, with this query: `SELECT PK FROM TBL1 WHERE col0 IN (SELECT col0 FROM TBL1)`